pkg/kubectl/cmd/util/factory.go

```diff
@@ -157,6 +158,7 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	flags.SetNormalizeFunc(util.WarnWordSepNormalizeFunc) // Warn for "_" flags
+	flags.Bool("disable-dynamic-discovery", false, "If true, disable calls to dynamically discover dynamic APIs")
```

Comment:
note, this isn't on client commands, its only on kubectl commands.  Kubectl commands require a human at the keyboard and are therefore a diminishingly small part of the overall load on the apiserver.